queuing_system/qinvestigate

```diff
@@ -510,8 +510,9 @@ amend_qstat() {
 		/^-----------/ { printf "%s  %s\n", $0, "-------------------"; next}
 		
 		#running or queuing job
-		/^[0-9]/ && $10 == "Q" {printf "%s  %s\n",$0,getstart($1); next}
-		/^[0-9]/ && $10 == "R" {printf "%s  %s\n",$0,getend($1); next}
+		{ a=$0; b=$1 }
+		/^[0-9]/ && $10 == "Q" {printf "%s  %s\n",a,getstart(b); next}
+		/^[0-9]/ && $10 == "R" {printf "%s  %s\n",a,getend(b); next}
```

Comment:
It feels weird to me to have `a` and `b` in global scope. How does
```awk
/^[0-9]/ && $10 == "Q" {job = $1; printf "%s  %s\n",$0,getstart(job); next}
```
or
```awk
/^[0-9]/ && $10 == "Q" {job = $1; line=$0; printf "%s  %s\n", line, getstart(job); next}
```
work? Also `a` and `b` are really bad variable names ...

Reply:
Actually, 

`/^[0-9]/ && $10 == "Q" {line = $0; printf "%s %s\n",line,getstart($1); next}`

works.

Reply:
Rather weird ... but fair enough. Do you have any clue why this is?

Reply:
Ah ... I see that makes sense to some extent, since `awk` rebuilds `$0` if one assigns to its "components" `$1` to `$NF` (see section 3.2 of [awk_course.pdf](https://michael-herbst.com/teaching/introduction-to-awk-programming-2016/introduction-to-awk-programming-2016/notes.pdf), last paragraph of page 14).

Given this insight, would it not be more reasonable to employ the first solution I proposed (in case it works), since `getstart` can than modify a copy of `$1` without effecting `$1` at all?

Reply:
Hm.  The first solution does not work, i. e. one still gets the erroneous output on startup:

```
/export/home/adempwol/dreuwBin/queuing_system/qinvestigate: Zeile 777: Warnung: Befehlsersetzung: Null Byte Eingabe ignoriert
You currently have no jobs.
```

and upon saying `joblist`:

```
Meister: 
                                                                                  Req'd    Req'd       Elap          Estimated
Job ID                  Username    Queue    Jobname          SessID  NDS   TSK   Memory   Time    S   Time     Start / Completion
----------------------- ----------- -------- ---------------- ------ ----- ------ ------ --------- - ---------  -------------------
Earliest completion in     1:59:58 on Tue Jan 29 17:24:27     16156     1      3  37580  02:00:00 R  00:00:12  Tue Jan 29 17:24:27
   Knecht18/8+Knecht18/9+Knecht18/10
Earliest completion in     2:00:00 on Tue Jan 29 17:24:29     16206     1      2  37580  02:00:00 R  00:00:10  Tue Jan 29 17:24:29
   Knecht18/11+Knecht18/12
```

The other way around, i. e. copying `line`, everything works as expected.  I still don't get what's the issue here.

Reply:
ok ... than we leave it to mystery.